### PR TITLE
feat(slice-9): recurring charges view

### LINKED ISSUE #10: Slice 9: Recurring charges view
## Parent

#1

## What to build

Surface recurring charges as a user-facing workflow so the product can help identify subscriptions and other repeat monthly obligations.

This slice should include:
- deterministic recurring detection using merchant similarity, amount tolerance, and cadence rules;
- a frontend view of recurring candidates with enough detail to be explainable;
- linkage from the recurring view back to supporting transactions;
- guardrails for false positives and sparse data.

Implementation notes:
- reuse the existing recurring detection backend where it matches the PRD, but complete the missing UI and product flow;
- keep the criteria explainable in-product so users can trust why something was flagged.

## Acceptance criteria

- [ ] Recurring candidates are computed deterministically from ledger data.
- [ ] The UI shows recurring candidates with merchant, typical amount, cadence, and occurrence count.
- [ ] Users can inspect the underlying supporting transactions for a recurring series.
- [ ] Edge cases with too few occurrences or excessive variance are handled predictably.
- [ ] Tests cover representative recurring and non-recurring scenarios plus the UI data contract.

## Blocked by

- #6
- #9

### PR DESCRIPTION
**Pull request by chandra-jagarlamudi**
Implements Slice 9: adds HTTP tests for recurring detection API and RecurringPage frontend showing merchant, amount, cadence, and occurrence count.

Closes #10

### CODE REVIEW

File: frontend/src/RecurringPage.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    <article className="panel recurring-row">
+      <div className="recurring-main">
+        <div className="recurring-merchant">
+          <strong>{charge.merchant}</strong>
+        </div>
+        <div className="recurring-amount">{formatAmount(charge.typical_amount)}</div>
+        <div className="recurring-occurrences">
+          <span className="metric-label">Occurrences</span>
+          {charge.occurrences} months
+        </div>
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply frontend/src/RecurringPage.tsx:34]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/36#discussion_r3176000251)

File: frontend/src/RecurringPage.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          {expanded ? 'Hide dates' : 'Show dates'}
+        </button>
+      </div>
+      {expanded && (
+        <ul className="recurring-dates">
+          {charge.monthly_dates.map((d) => (
+            <li key={d}>{d}</li>
+          ))}
+        </ul>
+      )}
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply frontend/src/RecurringPage.tsx:58]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/36#discussion_r3176000256)

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -163,10 +164,12 @@ function Shell({
         <nav className="shell-nav">
           <Link to="/">Overview</Link>
           <Link to="/smoke">API smoke</Link>
+          <Link to="/recurring">Recurring</Link>
         </nav>
         <Routes>
           <Route path="/" element={<Overview username={username} protectedState={protectedState} />} />
           <Route path="/smoke" element={<SmokePage protectedState={protectedState} />} />
+          <Route path="/recurring" element={<RecurringPage />} />
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply frontend/src/App.tsx:182]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/36#discussion_r3176000262)

File: backend/tests/test_recurring_http.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    b"2025-01-15,-50.00,GROCERY STORE\n"
+)
+
+
+def _ingest(client, account_id, csv_bytes, upload_dir):
+    return client.post(
+        "/ingest/csv",
+        data={"account_id": str(account_id)},
+        files={"file": ("stmt.csv", csv_bytes, "text/csv")},
+    )
```
[the remark this answers is not in the document]
Comment (chandra-jagarlamudi) [reply backend/tests/test_recurring_http.py]: @copilot apply changes based on [this feedback](https://github.com/chandra-jagarlamudi/PersonalFinancialAnalyst/pull/36#discussion_r3176000267)
